buildkite/client.go

```diff
@@ -32,21 +29,55 @@ type clientConfig struct {
 	userAgent  string
 }
 
+type headerRoundTripper struct {
+	next   http.RoundTripper
+	Header http.Header
+}
+
 // NewClient creates a client to use for interacting with the Buildkite API
 func NewClient(config *clientConfig) *Client {
-	token := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: config.apiToken})
-	httpClient := oauth2.NewClient(context.Background(), token)
+
+	// Setup a HTTP Client that can be used by all REST and graphql API calls,
+	// with suitable headers for authentication and user agent identification
+	var rt http.RoundTripper
+	rt = http.DefaultTransport
+	header := make(http.Header)
+	header.Set("Authorization", "Bearer "+config.apiToken)
+	header.Set("User-Agent", config.userAgent)
+	rt = newHeaderRoundTripper(rt, header)
+
+	httpClient := &http.Client{
+		Transport: rt,
+	}
 
 	return &Client{
 		graphql:      graphql.NewClient(config.graphqlURL, httpClient),
 		genqlient:    genqlient.NewClient(config.graphqlURL, httpClient),
 		http:         httpClient,
 		organization: config.org,
 		restUrl:      config.restURL,
-		userAgent:    config.userAgent,
 	}
 }
 
+func newHeaderRoundTripper(next http.RoundTripper, Header http.Header) *headerRoundTripper {
+	if next == nil {
+		next = http.DefaultTransport
+	}
+	return &headerRoundTripper{
+		next:   next,
+		Header: Header,
+	}
+}
+
+func (rt *headerRoundTripper) RoundTrip(req *http.Request) (resp *http.Response, err error) {
```

Comment:
Named returns are often considered harmful, also not used here.

```suggestion
func (rt *headerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
```